src/macis/gf/gf.cxx

```diff
@@ -0,0 +1,56 @@
+/*
+ * MACIS Copyright (c) 2023, The Regents of the University of California,
+ * through Lawrence Berkeley National Laboratory (subject to receipt of
+ * any required approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * See LICENSE.txt for details
+ */
+
+#include "macis/gf/gf.hpp"
+
+namespace macis {
+void write_GF(
+    const std::vector<std::vector<std::vector<std::complex<double> > > > &GF,
+    const std::vector<std::complex<double> > &ws,
+    const std::vector<int> &GF_orbs, const std::vector<int> &todelete,
+    const bool is_part) {
+  size_t nfreqs = ws.size();
+
+  if(GF_orbs.size() > 1) {
+    std::string fname = is_part ? "LanGFMatrix_ADD.dat" : "LanGFMatrix_SUB.dat";
+    std::ofstream ofile(fname);
+    ofile.precision(dbl::max_digits10);
+    for(int iii = 0; iii < nfreqs; iii++) {
+      ofile << std::scientific << real(ws[iii]) << " " << imag(ws[iii]) << " ";
+      for(int jjj = 0; jjj < GF[iii].size(); jjj++) {
+        for(int lll = 0; lll < GF[iii][jjj].size(); lll++)
+          ofile << std::scientific << real(GF[iii][jjj][lll]) << " "
+                << imag(GF[iii][jjj][lll]) << " ";
+      }
+      ofile << std::endl;
+    }
+    ofile.close();
+
+    std::string fname2 = is_part ? "GFMatrix_OrbitalIndices_ADD.dat"
+                                 : "GFMatrix_OrbitalIndices_SUB.dat";
+    std::ofstream ofile2(fname2);
+    for(int iii = 0; iii < GF_orbs.size(); iii++) {
+      if(std::find(todelete.begin(), todelete.end(), iii) != todelete.end())
+        continue;
+      ofile2 << GF_orbs[iii] << std::endl;
+    }
+    ofile2.close();
```

Comment:
`std::ofstream` is RAII - no need to explicitly close, it will close on destruction